journal.py

```diff
@@ -94,6 +104,15 @@ def add_entry(request):
         return HTTPInternalServerError
     return HTTPFound(request.route_url('home'))
 
+@view_config(route_name='update', request_method='POST')
+def update_entry(request):
+    identification = (request.matchdict.get('id',-1),)
+    try:
+        update(request, identification)
```

Comment:
Why extract the id from `request.matchdict` here, when you're just going to pass the request in to `update` anyway?  Can't you just get the id in that function when you need it?  `matchdict` isn't going to disappear from the request.